lib/commandCenter/commandCenter.js

```diff
@@ -388,15 +387,55 @@ function compress(content, cb) {
 	});
 }
 
+/**
+ * Promisify the user command execute method, or
+ * ensure its return value is passed to state.respond
+ *
+ * Once callback-based user commands are removed from MAGE,
+ * you will be able to remove this method.
+ *
+ * @param {*} cmdInfo
+ * @param {State} state
+ * @param {*} paramList
+ */
+function generateCommandPromise(cmdInfo, state, paramList) {
+	const mod = cmdInfo.mod;
+	const execute = () => mod.execute.apply(mod, paramList);
+
+	if (cmdInfo.isAsync) {
+		// Async user commands will need to take the response value, and feed it
+		// to `state.respond`
+		const isJson = mod.serialize === false;
+
+		return execute()
+			.then((response) => state.respond(response, isJson));
+	} else {
+		// Callback-based user commands are expected to call `state.respond`
+		// on their own; all we need to do is promisify
+		return new Promise((resolve) => execute(resolve));
+	}
+}
 
 /**
- * Sets the execution timeout of all user commands to the given duration
+ * Create a promise timer
  *
- * @param {number} timeout   Timeout in milliseconds
+ * Useful when using Promise.race.
+ *
+ * @param {number} timeout
  */
-CommandCenter.prototype.setUserCommandTimeout = function (timeout) {
-	this.userCommandTimeout = timeout;
-};
+function createTimerPromise(timeout) {
+	let timer;
+
+	const promise = new Promise((resolve, reject) => {
+		if (timeout) {
+			timer = setTimeout(() => reject(new Error('timeout')), timeout);
```

Comment:
I think you'll want to add the user command name or state description value to the error message here. Else, when a timeout happens, you won't know on what, right?

Reply:
`state.error` logs description - which is set to the name of the user command.